lib/internal/http2/compat.js

```diff
@@ -84,8 +84,8 @@ let statusConnectionHeaderWarned = false;
 // close as possible to the current require('http') API
 
 const assertValidHeader = hideStackFrames((name, value) => {
-  if (name === '' ||
-      typeof name !== 'string' ||
+  if (typeof name !== 'string' ||
+      name === '' ||
```

Comment:
Unrelated change?

Reply:
> Validate the type of the value before comparing.

That change was made for that reason, but it doesn't affect anything much, in most places we validate the type first before doing comparisons or other operations on the values but I could revert that change if you want.

Reply:
For this kind of change, I'm -0 if we don't have a linter rule to enforce it. But I also don't think having a linter rule for that would be worth it 🤷‍♂️

Reply:
Honestly, it's just easier to read the original version since each line is shorter than the next. Also -0 on this particular change.